homeassistant/components/google/__init__.py

```diff
@@ -262,6 +254,14 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     )
     hass.data[DOMAIN][entry.entry_id][DATA_SERVICE] = calendar_service
 
+    if entry.unique_id is None:
+        try:
+            primary_calendar = await calendar_service.async_get_calendar("primary")
+        except ApiException as err:
+            _LOGGER.debug("Error reading calendar primary calendar: %s", err)
```

Comment:
Should we raise `ConfigEntryNotReady` here? If we don't update the config entry unique_id the state of the integration will be inconsistent.

Reply:
Sure, removed gradeful degradation and now there is a how hard fail instead.